pretyping/constr_matching.ml

```diff
@@ -235,6 +235,9 @@ let matches_core env sigma convert allow_partial_app allow_bound_rels
 
       | PMeta None, m -> subst
 
+      | PEvar (ev, arg1), Evar (ev', arg2) when Evar.equal ev ev' ->
+         Array.fold_left2 (sorec ctx env) subst arg1 arg2
+
```

Comment:
This has already been independently solved in trunk, and probably triggers a spurious branch warning.

Reply:
Indeed, I killed that commit.